tests/app/controllers/__init__.py

```diff
@@ -102,15 +102,16 @@ def _verify_item(cls, item, attrs=None):
         # database before running this test, the `db_*_model_initialized` fixtures take care of it.
         assert item["id"] == 1
         for key, value in attrs.items():
-            try:
-                subcls, _ = cls.attrs[key]
-            except (TypeError, ValueError):
-                pass
-            else:
-                if isinstance(subcls, type) and issubclass(subcls, BaseTestController):
-                    # skip verifying objects this one depends on
-                    continue
-            assert item[key] == value
+            if key != "api_key":
+                try:
+                    subcls, _ = cls.attrs[key]
+                except (TypeError, ValueError):
+                    pass
+                else:
+                    if isinstance(subcls, type) and issubclass(subcls, BaseTestController):
+                        # skip verifying objects this one depends on
+                        continue
+                assert item[key] == value
```

Comment:
Uhh, we shouldn't special case things like this in `BaseTestController`. I have a similar requirement to have fields submitted in POST that shouldn't be verified in the result, let me carve that out of the heap of changes in my branch and put it in here.

Reply:
As discussed, there's `BaseTestController.no_response_attrs` now which can be set in child classes (which I did with the latest push).